apps/epic-react/src/lib/modules.ts

```diff
@@ -1,105 +1,186 @@
 import groq from 'groq'
 import {sanityClient} from '@skillrecordings/skill-lesson/utils/sanity-client'
+import {z} from 'zod'
+
+export const ModuleSchema = z.object({
+  _id: z.string(),
+  _type: z.string(),
+  title: z.string(),
+  slug: z.object({
+    _type: z.literal('slug'),
+    current: z.string(),
+  }),
+  state: z.string(),
+  moduleType: z.string(),
+  description: z.string().nullable(),
+  github: z
+    .object({
+      repo: z.string(),
+    })
+    .nullable(),
+  square_cover_large_url: z.string(),
+  resources: z.array(
+    z.object({
+      _id: z.string(),
+      _type: z.enum(['section', 'exercise', 'explainer', 'interview']),
+      title: z.string(),
+      slug: z.string(),
+      _updatedAt: z.string(),
+      _createdAt: z.string(),
+      description: z.string().nullable(),
+      resources: z
+        .array(
+          z.object({
+            _id: z.string(),
+            _type: z.enum(['exercise', 'explainer', 'interview']),
+            title: z.string(),
+            slug: z.string(),
+            _updatedAt: z.string(),
+            _createdAt: z.string(),
+            description: z.string().nullable(),
+            body: z.array(z.any()),
+          }),
+        )
+        .optional(),
+      body: z.array(z.any()).optional(),
+    }),
+  ),
+})
+
+export const ProductModulesSchema = z.object({
+  _id: z.string(),
+  title: z.string(),
+  description: z.string(),
+  modules: z.array(ModuleSchema),
+})
+
+const productModulesQuery = groq`*[_type == "product" && productId == $productId][0] {
```

Comment:
annotated screenshot of how this exposes the nested structure of sections and lessons for each module:

![CleanShot 2023-04-25 at 08 18 04@2x](https://user-images.githubusercontent.com/694063/234274467-237f29cb-cb0f-46f2-8a39-ed687ef936cb.png)